src/plugins/discover/public/application/components/sidebar/discover_field_details.tsx

```diff
@@ -39,6 +39,18 @@ export function DiscoverFieldDetails({
   onAddFilter,
 }: DiscoverFieldDetailsProps) {
   const warnings = getWarnings(field);
+  const [showVisualizeLink, setShowVisualizeLink] = useState<boolean | undefined>(false);
+
+  useEffect(() => {
+    isFieldVisualizable(field, indexPattern.id, details.columns).then(
```

Comment:
nit: makes sense to handle cancelations to prevent possible race conditions. Not sure though if they can occur in this particular case